server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -785,20 +781,20 @@ public IndexMetadata getIndexSafe(Index index) {
         throw new IndexNotFoundException(index);
     }
 
-    public ImmutableOpenMap<String, IndexMetadata> indices() {
-        return this.indices;
+    public Map<String, IndexMetadata> indices() {
+        return Collections.unmodifiableMap(this.indices);
```

Comment:
Can you store the fields as `Collections.unmodifiableMap` at instance creation time to ensure that these are never mutated and that a mutable map is never accidentally exposed?

Reply:
yes! And doing that will also keep from creating a new map everytime the getter is called. Good catch!

Reply:
Done